packages/server/src/question/question.subscriber.ts

```diff
@@ -62,6 +63,29 @@ export class QuestionSubscriber
   }
 
   async afterInsert(event: InsertEvent<QuestionModel>): Promise<void> {
+    const numberOfQuestions = await QuestionModel.openInQueue(
+      event.entity.queueId,
+    )
+      .where('question.status IN (:...openStatus)', {
+        openStatus: [OpenQuestionStatus.Drafting, OpenQuestionStatus.Queued],
+      })
+      .getCount();
+
+    const staff = (
+      await QueueModel.findOne(event.entity.queueId, {
+        relations: ['staffList'],
+      })
+    ).staffList;
+
+    if (numberOfQuestions === 0) {
```

Comment:
won't this never notify since this is run after insert? i'm not sure how this subscriber works, but i'm not getting the notification when testing

Reply:
DM'd, but in my testing (console logs) I was able to see that there were no questions here, and then I got a notif only on the first student.

Reply:
survey says it works on local but not vercel 👊

Reply:
It's because we're not running inside the same db transaction as the insertion, so we see the state of the db before insertion